askama/src/filters/default.rs

```diff
@@ -0,0 +1,403 @@
+impl<T: DefaultFilterable + alloc::borrow::ToOwned + ?Sized> DefaultFilterable
+    for alloc::borrow::Cow<'_, T>
+{
+    type Filtered<'a>
+        = T::Filtered<'a>
+    where
+        Self: 'a;
+
+    type Error = T::Error;
+
+    #[inline]
+    fn as_filtered(&self) -> Result<Option<Self::Filtered<'_>>, Self::Error> {
+        self.as_ref().as_filtered()
+    }
+}
+
+/// A [`Wrapping`] integer has a value if it is not `0`.
+impl<T: DefaultFilterable> DefaultFilterable for Wrapping<T> {
+    type Filtered<'a>
+        = T::Filtered<'a>
+    where
+        Self: 'a;
+
+    type Error = T::Error;
+
+    #[inline]
+    fn as_filtered(&self) -> Result<Option<Self::Filtered<'_>>, Self::Error> {
+        self.0.as_filtered()
+    }
+}
+
+/// A [`Saturating`] integer has a value if it is not `0`.
+impl<T: DefaultFilterable> DefaultFilterable for Saturating<T> {
+    type Filtered<'a>
+        = T::Filtered<'a>
+    where
+        Self: 'a;
+
+    type Error = T::Error;
+
+    #[inline]
+    fn as_filtered(&self) -> Result<Option<Self::Filtered<'_>>, Self::Error> {
+        self.0.as_filtered()
+    }
+}
+
+macro_rules! impl_for_int {
+    ($($ty:ty)*) => { $(
+        #[doc = concat!("A [`", stringify!($ty), "`] has a value if it is not `0`.")]
+        impl DefaultFilterable for $ty {
+            type Filtered<'a> = $ty;
+            type Error = Infallible;
+
+            #[inline]
+            fn as_filtered(&self) -> Result<Option<$ty>, Infallible> {
+                match *self {
+                    0 => Ok(None),
+                    value => Ok(Some(value)),
+                }
+            }
+        }
+    )* };
+}
+
+impl_for_int!(
+    u8 u16 u32 u64 u128 usize
+    i8 i16 i32 i64 i128 isize
+);
+
+macro_rules! impl_for_non_zero {
+    ($($name:ident : $ty:ty)*) => { $(
+        #[doc = concat!("A [`", stringify!($name), "`][num::", stringify!($name),"] always has a value.")]
+        impl DefaultFilterable for num::$name {
+            type Filtered<'a> = $ty;
+            type Error = Infallible;
+
+            #[inline]
+            fn as_filtered(&self) -> Result<Option<$ty>, Infallible> {
+                Ok(Some(self.get()))
+            }
+        }
+    )* };
+}
+
+impl_for_non_zero!(
+    NonZeroU8:u8 NonZeroU16:u16 NonZeroU32:u32 NonZeroU64:u64 NonZeroU128:u128 NonZeroUsize:usize
+    NonZeroI8:i8 NonZeroI16:i16 NonZeroI32:i32 NonZeroI64:i64 NonZeroI128:i128 NonZeroIsize:isize
+);
+
+/// A `bool` has a value if it is [`true`].
+impl DefaultFilterable for bool {
+    type Filtered<'a> = bool;
+    type Error = Infallible;
+
+    #[inline]
+    fn as_filtered(&self) -> Result<Option<bool>, Infallible> {
+        match *self {
+            true => Ok(Some(true)),
+            false => Ok(None),
+        }
+    }
+}
+
+/// An `f32` has a value if it is [`Normal`][FpCategory::Normal], i.e. it is not zero,
+/// not sub-normal, not infinite and not NaN.
+impl DefaultFilterable for f32 {
+    type Filtered<'a>
+        = Self
```

Comment:
Wow this formatting is super weird. O.o

Reply:
Yeah, it's not the formatting I would have preferred, but rustfmt insists on adding a linebreak there. :)